src/components/MarkupToolbar.tsx

```diff
@@ -14,10 +14,41 @@ import {
 } from "../utils/codeMirrorExtensions";
 import {ClozeQuestionContext} from "./semantic/presenters/ItemQuestionPresenter";
 
-export const MarkupToolbar = ({codemirror, encoding}: { codemirror: RefObject<ReactCodeMirrorRef>, encoding: string | undefined }) => {
+export const MarkupToolbar = ({set, cancel, codemirror, encoding}: { set: () => void, cancel: () => void, codemirror?: RefObject<ReactCodeMirrorRef>, encoding: string | undefined }) => {
     const inClozeQuestion = useContext(ClozeQuestionContext);
-    if (isMarkupEncoding(encoding)) {
-        return <div className={"d-flex w-100 bg-light border-bottom p-1 " + styles.cmMenuBar}>
+
+    const [wide, setWide] = useState(true);
+    const toolbarRef = useRef<HTMLDivElement>(null);
+
+    const updateIsWide = useCallback(() => {
+        const MIN_WIDTH = 500;
+        const width = toolbarRef.current?.clientWidth;
+        setWide(!width || width >= MIN_WIDTH);
+    }, []);
+
+    useLayoutEffect(() => {
+        updateIsWide();
+    }, []);
+
+    useEffect(() => {
+        if (!toolbarRef.current) return;
+        const resizeObserver = new ResizeObserver(updateIsWide);
+        resizeObserver.observe(toolbarRef.current);
+        return () => resizeObserver.disconnect();
+    }, []);
+
+    return <div ref={toolbarRef} className={"d-flex w-100 bg-light border-bottom p-1 " + styles.cmMenuBar}>
+        <button className={styles.cmPanelButton} title={"Set (Mod-Shift-Enter)"}
+                aria-label={"Save changes to this markup block (shortcut is Mod-Shift-Enter)"}
+                onClick={set}>
+            💾{wide && " Set"}
+        </button>
+        <button className={styles.cmPanelButton} title={"Cancel (Esc)"}
```

Comment:
This is harder than it seems because the "Set" and "Cancel" buttons are inside `buildValuePresenter` - we can't add the shortcut prompt to those buttons because they aren't valid for any value presenters apart form `BaseValuePresenter`